aten/src/ATen/DeviceAccelerator.cpp

```diff
@@ -1,39 +1,37 @@
-#include <ATen/DeviceAccelerator.h>
 #include <ATen/Context.h>
-
+#include <ATen/DeviceAccelerator.h>
 namespace at {
 
 C10_API std::optional<DeviceType> getAccelerator(bool checked) {
-#define CHECK_NO_CUDA \
-  TORCH_CHECK(!at::hasCUDA(), "Cannot have both CUDA and PrivateUse1");
-
-#define CHECK_NO_PU1 \
-  TORCH_CHECK(!is_privateuse1_backend_registered(), "Cannot have both CUDA and PrivateUse1");
-
-#define CHECK_NO_MTIA \
-  TORCH_CHECK(!at::hasMTIA(), "Cannot have MTIA with other devices");
-
-    if (is_privateuse1_backend_registered()) {
-        // We explicitly allow PrivateUse1 and another device at the same time
-        // as we use this for testing.
-        // Whenever a PrivateUse1 device is registered, use it first.
-        return kPrivateUse1;
-    } else if (at::hasCUDA()) {
-        CHECK_NO_PU1
-        CHECK_NO_MTIA
-        return kCUDA;
-    } else if (at::hasMTIA()) {
-        CHECK_NO_CUDA
-        CHECK_NO_PU1
-        return kMTIA;
-    } else {
-        TORCH_CHECK(!checked, "Cannot access accelerator device when none is available.")
-        return std::nullopt;
-    }
-
-#undef CHECK_NO_CUDA
-#undef CHECK_NO_PU1
+#define ASSIGN_ACCELERATOR_AND_CHECK_MUTEX(device_name) \
+  if (at::has##device_name()) {                         \
+    device_type = k##device_name;                       \
+    TORCH_CHECK(                                        \
+        !is_mutex_device_detected,                      \
```

Comment:
Do you mean "mutex" or "multiple" here?

Reply:
I meat `mutex`, but `multiple` is better. I will update to `multiple`.

Reply:
I go through the code and think that `mutex` is better. It means that if the mutually exclusive device is detected. We assign `is_mutex_device_detected` to `True` when the first accelerator is detected. And check it at the sequent accelerator detection to ensure the mutual exclusiveness in PyTorch.

Reply:
What you said makes sense. To avoid misleading in the name, I change `ASSIGN_ACCELERATOR_AND_CHECK_MUTEX` to `DETECT_AND_ASSIGN_ACCELERATOR` and `is_mutex_device_detected` to `is_accelerator_detected`.